src/NServiceBus.RavenDB/Timeouts/QueryTimeouts.cs

```diff
@@ -83,8 +89,18 @@ public QueryTimeouts(IDocumentStore store)
             return new TimeoutsChunk(results, nextTimeToRunQuery);
         }
 
+        internal void Shutdown()
+        {
+            abort = true;
+        }
+
         public async Task<IEnumerable<TimeoutsChunk.Timeout>> GetCleanupChunk(DateTime startSlice)
         {
+            if(abort)
+            {
+                return new TimeoutsChunk.Timeout[ 0 ];
```

Comment:
In this case I'd prefer to leave it as is since the query is executed and not streamed, leaving the check in place adds an option to skip useless work if the endpoint is shutting down